doc/rbd/rbd-live-migration.rst

```diff
@@ -91,6 +108,181 @@ usage during the migration process::
+            "image_name": "<image-name>",
+            ["image_id": "<image-id>",] (optional if image in trash)
+            "snap_name": "<snap-name>",
+            ["snap_id": "<snap-id>",] (optional alternative to "snap_name")
+        }
+
+Note that the ``native`` format does not include the ``stream`` object since
+it utilizes native Ceph operations. For example, to import from the image
+``rbd/ns1/image1@snap1``, the ``source-spec`` could be encoded as::
+
+        {
+            "type": "native",
+            "pool_name": "rbd",
+            "pool_namespace": "ns1",
+            "image_name": "image1",
+            "snap_name": "snap1"
+        }
+
+The ``raw`` format can be used to describe a thick-provisioned, raw block device
+export (i.e. `rbd export --export-format 1 <snap-spec>`). The ``raw`` format
+data can be linked to any supported stream source described below. For example,
+its base ``source-spec`` JSON is encoded as follows::
+
+        {
+            "type": "raw",
+            "stream": {
+                <stream unique parameters for HEAD, non-snapshot revision>
+            },
+            "snapshots": [
+                {
+                    "type": "raw",
+                    "name": "<snapshot-name>",
+                    "stream": {
+                        <stream unique parameters for snapshot>
+                    }
+                },
+            ] (optional oldest to newest ordering of snapshots)
+        }
+
+Support for snapshots is optional and currently only supports thick-provisioned
+``raw`` snapshot exports.
+
+Additional formats such as QCOW2, RBD export-format v2, and RBD export-diff
+snapshots will be added in a future release.
+
+Streams
+~~~~~~~
+
+The ``file`` stream can be used to import from a locally accessible POSIX file
+source. Its ``source-spec`` JSON is encoded as follows::
+
+        {
+            <format unique parameters>
+            "stream": {
+                "type": "file",
+                "file_path": "<file-path>"
+            }
+        }
+
+For example, to import a raw-format image from a file located at
+"/mnt/image.raw", its ``source-spec`` JSON is encoded as follows::
+
+        {
+            "type": "raw",
+            "stream": {
+                "type": "file",
+                "file_path": "/mnt/image.raw"
+            }
+        }
+
+The ``http`` stream can be used to import from a remote HTTP(s) web server.
+Its ``source-spec`` JSON is encoded as follows::
+
+        {
+            <format unique parameters>
+            "stream": {
+                "type": "http",
+                "url": "<url-path>"
+            }
+        }
+
+For example, to import a raw-format image from a file located at
+"http://download.ceph.com/image.raw", its ``source-spec`` JSON is encoded
+as follows::
+
+        {
+            "type": "raw",
+            "stream": {
+                "type": "http",
+                "url": "http://download.ceph.com/image.raw"
+            }
+        }
```

Comment:
nit: probably it is worth mentioning that "url" supports both "http" and "https" schemes, or have "https" in one of examples.

Reply:
I had noted it was `HTTP(s)` above but I'll separate that to make it clearer.